### DISCUSSION TITLE
Not Able to Send files from Phone to Desktop

### DISCUSSION THREAD

**Discussion by Bill4Riding** [localsend/localsend#1075] [Discussion (English, International)]
Have Local Send on three machines.  It is on my Windows 10 desktop, my Samsung s23 phone, and on Linux Mint, latest version.
Can send and receive files on both my Windows machine and Samsung phone.

However, on my Samsung phone and Linux Mint desktop, I can ONLY send files from the desktop to the phone.     Not able to send files from the phone to the Linux Mint machine.    It keeps waiting for a response and then it times out, producing an error.

Not able to correct the issue of not being able to send files from the phone to the Linux machine.    But it does work from the Linux machine to the phone.     Does anyone have any ideas?   Did uninstall Local Send from Linux Mint and then reinstalled it.   But that did not work.

**Comment by Tienisto** [comment]
This is likely a firewall error

**Reply by Bill4Riding** [reply]
Thank you, Tienisto.  You were right.   It was the firewall.   At the moment, I need to turn the firewall off in order to use Local Send.   I trying to create a RULE to allow Local Send to work.   However, I have not been successful in doing so.   I will keep on trying.   At some point I'll figure it out.   Again thanks for pointing me in the right direction.

**Reply by predators9** [reply]
> This is likely a firewall error

Thank You

**Comment by kgn2015** [comment]
I created this rule in Gufw firewall (in "Simple" tab) and it works for me:
```
Name: LocalSend-app
Policy: Allow
Direction: Both (IN & OUT)
Protocol: Both (TCP & UDP)
Port: 53317
```

**Reply by Hugh-Sala** [reply]
It worked for me as well. The port was missing on my Debian 12 firewall... to transfer from Android to Debian and vice-versa. But I'm still struggling to connect to Windows 11...

**Reply by pamodmalith** [reply]
It worked for me as well.

**Reply by bronxvoltaics** [reply]
+1 on making a new Simple rule for port 53317

**Reply by PierreHvrd** [reply]
Worked for me too, thanks !

**Reply by ShriSingh** [reply]
So my firewall type is 'ufw' and I'm not able to find localsend app when setting up the rule:
<img width="798" height="430" alt="Screenshot_20260413_202543" src="https://github.com/user-attachments/assets/bab92172-4c70-4d0e-b88d-b6666a2e3d1c" />
Do I need to manually enter the ip address value and port number to set the exception then?

**Reply by harshv5094** [reply]
You can just enter the port number (default is 53317)
```bash
sudo ufw allow 53317/tcp comment "LocalSend-App" && sudo ufw allow 53317/udp comment "LocalSend-app"
```

**Reply by RamonP12** [reply]
> You can just enter the port number (default is 53317)
> 
> ```shell
> sudo ufw allow 53317/tcp comment "LocalSend-App" && sudo ufw allow 53317/udp comment "LocalSend-app"
> ```

This works for me to solve exactly this problem in CachyOS.
Before: Sending from CachyOS to Samsung mobile Devices did work, vice versa did not. 
Now: Both directions work. 

But is it not a risk to open this port in general? For me it seems the rule opens the port for any kind of requests not only specific for this app?

**Reply by harshv5094** [reply]
@RamonP12 you can use `limit` to limit port connectivity instead of denying it, Naming your port "Localsend" allows me to quickly turn off/on ports for the app.

```bash
# Limit port access
sudo ufw limit "LocalSend"

# Allow Port access
sudo ufw allow "LocalSend"
```

So you can rest assured that no one can access your port

<img width="1366" height="768" alt="image" src="https://github.com/user-attachments/assets/27cf50b8-1f71-4afe-87d2-56c91023b8f0" />

**Reply by Bulbofsalad** [reply]
> I created this rule in Gufw firewall (in "Simple" tab) and it works for me:
> 
> ```
> Name: LocalSend-app
> Policy: Allow
> Direction: Both (IN & OUT)
> Protocol: Both (TCP & UDP)
> Port: 53317
> ```

tysm this worked instantly. I didnt know linux would throw a similar issue but i guess it makes sense when dealing with this kind of stuff ty again ^^

**Comment by Bill4Riding** [comment]
Thanks, I will try it out. 

Bill 

⁣Get BlueMail for Android ​

**Comment by buzzkirill** [comment]
Hi, I'm also having this problem - receiving files on a Windows 10 Desktop from a Samsung Galaxy phone only works if the firewall is turned off. Allowing the connection for the Localsend app in the fiewall rules doesn't work. Can anyone provide a "rule" solution? bit of a noob here

**Reply by ngogb02** [reply]
Hey, this is what I did and it worked for me. Go to firewall > rules > select "Simple" tab > for name, I just put in "LocalSend", then at the port. Go to LocalSend app on phone, go to advanced settings, look at the port, it should be some type of number. Input that same number in the port on LocalSend Simple rule firewall on PC. Select direction, as appropriate.

**Reply by buzzkirill** [reply]
> Hey, this is what I did and it worked for me. Go to firewall > rules > select "Simple" tab > for name, I just put in "LocalSend", then at the port. Go to LocalSend app on phone, go to advanced settings, look at the port, it should be some type of number. Input that same number in the port on LocalSend Simple rule firewall on PC. Select direction, as appropriate.

Sorry, I can't find the setting you're referring to on my Windows 10. Could you please be a bit more specific how to find it?

**Reply by MasterGreed** [reply]
> > Hey, this is what I did and it worked for me. Go to firewall > rules > select "Simple" tab > for name, I just put in "LocalSend", then at the port. Go to LocalSend app on phone, go to advanced settings, look at the port, it should be some type of number. Input that same number in the port on LocalSend Simple rule firewall on PC. Select direction, as appropriate.
> 
> Sorry, I can't find the setting you're referring to on my Windows 10. Could you please be a bit more specific how to find it?

I don't know if you still have that problem, but thanks to the previous guy it fixed my issue, basically, you go to your firewall > New rule > Port > Leave it at TCP and in " specific local ports" write the port that you'll find in your localsend app in additional settings > Just write LocalSend in the name, i hope this helps

**Comment by ghost** [comment]
Same problem here in Linux. The only thing that's bad is the inability to force things and the giant context that localsend provides when it draws:

`error`.

**Reply by jamestjw** [reply]
``` 
sudo firewall-cmd --zone=public --permanent --add-port=53317/tcp
sudo firewall-cmd --zone=public --permanent --add-port=53317/udp

firewall-cmd --reload
```
does this work for you?

**Reply by amphyvi** [reply]
Worked perfectly with `firewalld`, I was able to deduce the first 2 lines but I didn't think to try `firewall-cmd --reload`. You're a lifesaver!

**Comment by ebdulrehmandemya** [comment]
Another thing you should consider is reseting your OS DNS to default . Otherwise sending doesn't work.

**Comment by TallenPeli** [comment]
I just did `systemctl disable firewalld` and then enabled it again when the file was done transferring

**Comment by nnarain** [comment]
I didn't have any luck with changing firewall rules on Windows 10. Changing the port used is what worked for me:
https://github.com/localsend/localsend/issues/125#issuecomment-1537544190

**Reply by Amir10i** [reply]
I did it too and it worked

**Comment by fazzaan** [comment]
For some reason the firewall in Garuda (Arch-based) didn't need configuring, but in CachyOS (also Arch-based) it did.
@kgn2015 's solution worked.
https://github.com/localsend/localsend/discussions/1075#discussioncomment-8315105 

Can LocalSend open the firewall port by itself, or have a button or info built into the troubleshooter to fix it?

**Comment by harshv5094** [comment]
Here is the same command by using `ufw` for future visitor:
```sh
sudo ufw allow 53317/tcp comment "LocalSend-App" && sudo ufw allow 53317/udp comment "LocalSend-app"
```

Note: If you are using different port in Localsend, just replace it with the port number mentioned in this example
